pkg/data/examples.go

```diff
@@ -2462,96 +2503,118 @@ func createExampleAppWithGroups2() *Example {
+	"foo-frontend":         {"New-VM-3"},
+	"research-test-expr-2": {"New-VM-1"},
+}
+var policiesAppWithGroupsAndSegments = []Category{
+	{
+		Name:         "foo-app",
+		CategoryType: "Application",
+		Rules: []Rule{
+			{
+				Name:     "foo-allow-http-to-backend",
+				ID:       1027,
+				Source:   "foo-frontend",
+				Dest:     "foo-backend",
+				Services: []string{"/infra/services/HTTP"},
+				Action:   Allow,
+			},
+			{
+				Name:     "default-deny-foo-app",
+				ID:       1028,
+				Source:   "foo-app",
+				Dest:     "foo-app",
+				Services: []string{AnyStr},
+				Action:   Drop,
 			},
 		},
-		{
-			Name:         "New Policy",
-			CategoryType: "Application",
-			Rules: []Rule{
-				{
-					Name:     "allow-smb-to-foo-frontend",
-					ID:       1025,
-					Source:   "research-test-expr-2",
-					Dest:     "foo-frontend",
-					Services: []string{"/infra/services/SMB"},
-					Action:   Allow,
-				},
-				{
-					Name:     "allow-bar-app-https",
-					ID:       1024,
-					Source:   "bar-app",
-					Dest:     "bar-app",
-					Services: []string{"/infra/services/HTTPS"},
-					Action:   Allow,
-				},
+	},
+	{
+		Name:         "New Policy",
+		CategoryType: "Application",
+		Rules: []Rule{
+			{
+				Name:     "allow-smb-to-foo-frontend",
+				ID:       1025,
+				Source:   "research-test-expr-2",
+				Dest:     "foo-frontend",
+				Services: []string{"/infra/services/SMB"},
+				Action:   Allow,
+			},
+			{
+				Name:     "allow-bar-app-https",
+				ID:       1024,
+				Source:   "bar-app",
+				Dest:     "bar-app",
+				Services: []string{"/infra/services/HTTPS"},
+				Action:   Allow,
 			},
 		},
-		{
-			Name:         "Default Layer3 Section",
-			CategoryType: "Application",
-			Rules: []Rule{
-				{
-					Name:     "deny-research-app",
-					ID:       1021,
-					Source:   "research-app",
-					Dest:     "research-app",
-					Services: []string{AnyStr},
-					Action:   Drop,
-				},
-				DefaultDenyRule(2),
+	},
+	{
+		Name:         "Default Layer3 Section",
+		CategoryType: "Application",
+		Rules: []Rule{
+			{
+				Name:     "deny-research-app",
+				ID:       1021,
+				Source:   "research-app",
+				Dest:     "research-app",
+				Services: []string{AnyStr},
+				Action:   Drop,
 			},
+			DefaultDenyRule(2),
 		},
 	},
-})
+}
+
+var ExampleAppWithGroupsAndSegmentsBasic = Example{
+	Name:          "ExampleAppWithGroupsAndSegments",
+	VMs:           vmsAppWithGroupsAndSegments,
+	VMsAddress:    vmsAddressesWithGroupsAndSegments,
+	SegmentsByVMs: segmentsByVMsAddressesWithGroupsAndSegments,
+	SegmentsBlock: segmentsBlockAppWithGroupsAndSegments,
+	SegmentsT1GWs: segmentsT1GWsAppWithGroupsAndSegments,
+	GroupsByVMs:   groupsByVMsAppWithGroupsAndSegments,
+	Policies:      policiesAppWithGroupsAndSegments,
+}
+
+var ExampleAppWithGroupsAndSegments = registerExample(getExampleAppWithGroupsAndSegmentsInfer(false))
+
+var ExampleAppWithGroupsAndSegmentsForInfer = registerExample(getExampleAppWithGroupsAndSegmentsInfer(true))
```

Comment:
why do we need two examples objects?
should only define here `ExampleAppWithGroupsAndSegments` , and in `synthesis_test`, use this example to run once with `infer` flag and once without it.